### PR TITLE
[Pager] Allow HorizontalPager to wrap content height and VerticalPager to wrap content width

### LINKED ISSUE #753: HorizontalPager from 0.19 no more support wrapHeight
## Describe the bug

Starting with accompanist 0.19 `HorizontalPager` no more wrap to it's content height even when forcing it on the modifier.
It worked previously.
Forcing an `aspectRatio()` works but does not allow proper wrapping.

## To Reproduce
Create an `HorizontalPager` with  small height content and try to have the pager to wrap the pager height to that content.

## Expected behavior
HorizontalPager should be able to wrap to content height

## Environment:
 - Android OS version: Tested on api 30 / 31
 - Device: Any
 - Accompanist version: 0.19

### LINKED ISSUE #779: HorizontalPager 0.19.0 enforce verticalAlignment
## Describe the bug
The verticalAlignment parameters is not working with the HorizontalPager (and the horizontalAlignment for the VerticalPager might not be working too)

## To Reproduce
```
@OptIn(ExperimentalPagerApi::class)
@Composable
fun PagerContent() {
    HorizontalPager(
        count = 10,
        contentPadding = PaddingValues(start = 20.dp, end = 20.dp),
        itemSpacing = 16.dp,
        verticalAlignment = Alignment.Top,
        modifier = Modifier
            .fillMaxSize()
            .background(Color.Green)
    ) {
        Box(
            Modifier
                .requiredHeight(100.dp)
                .background(Brush.verticalGradient(colors = listOf(Color.Blue, Color.Red)))
                .fillMaxWidth()
        )
    }
}
```
This code should display a box of 100.dp at the top of the pager. Instead they are centered.

## Expected behavior
The item of the pager should be vertically aligned.

## Screenshots?

Expected :
image
What we got now : 
image

## Environment:
 - Accompanist version: 0.19.0
 - Compose version : 1.0.3 and 1.1.0-alpha05

## Additional context
The issue seems to be that deep down :

```kotlin
items(
                count = count,
                key = key,
            ) { page ->
                Box(
                    Modifier
                        // We don't any nested flings to continue in the pager, so we add a
                        // connection which consumes them.
                        // See: https://github.com/google/accompanist/issues/347
                        .nestedScroll(connection = ConsumeFlingNestedScrollConnection)
                        // Constraint the content to be <= than the size of the pager.
                        .fillParentMaxSize()
                        .wrapContentSize()
                ) {
                    pagerScope.content(page)
                }
            }
```

Each items of our pager is wrapped into a Box taking the full size of the parent, meaning the verticalAlignement cannot behave correctly here.

Not sure what the best way to handle this should be. Maybe just passing a Box Alignement through the pager ?

### PR DESCRIPTION
**Pull request by dmitrytavpeko**
Fixes #753
Fixes #779

### CODE REVIEW

**Comment by rrocav3** [comment]
@dmitrytavpeko  Is there any news about it? Thanks!

**Comment by dmitrytavpeko** [comment]
@rrocav3 Waiting for review by maintainers

**Comment by dmitrytavpeko** [comment]
Fixes #779 as well

**Review by andkulikov — APPROVED** [review]

**Pull request merged**
